### PR TITLE
Notify Github about wimpy deployments

### PR DESCRIPTION
**Pull request by fiunchinho**
The deployments should now appear on the Pull request information on Github UI.

This only covers the happy path for now. If we think it's interesting, we can try to cover when deployments fail

### CODE REVIEW

**Comment by fiunchinho** [comment]
@Hyunk3l @aramirez-es @ismFerDev

File: defaults/main.yml [hunk 1]
```diff
@@ -118,6 +119,16 @@ wimpy_aws_elb_stickiness:
...
 wimpy_aws_instance_type: t2.small # EC2 instance type
 
+# Github
+github_url_parts: "{{ wimpy_github_project_url.split('/') }}"
+github_proto: "{{ github_url_parts[0] | default('') | regex_replace(':', '') }}"
+github_host: "{{ github_url_parts[2] | default('')  }}"
+github_vendor: "{{ github_url_parts[3] | default('')  }}"
+github_name: "{{ github_url_parts[4] | default('') }}"
+github_project: "{{ github_vendor ~ '/' ~ github_name }}"
+github_api: "{{ ( github_host == 'github.com' ) | ternary('https://api.github.com','https://' ~ github_host ~ '/api/v3') }}"
```
Comment (Hyunk3l) [inline defaults/main.yml]: should cover both github.com and ghe, right?

File: defaults/main.yml [hunk 2]
```diff
@@ -118,6 +119,16 @@ wimpy_aws_elb_stickiness:
 
 wimpy_aws_instance_type: t2.small # EC2 instance type
 
+# Github
+github_url_parts: "{{ wimpy_github_project_url.split('/') }}"
+github_proto: "{{ github_url_parts[0] | default('') | regex_replace(':', '') }}"
```
Comment (Hyunk3l) [inline defaults/main.yml]: unused?

File: tasks/deploy.yml [hunk 3]
```diff
@@ -13,4 +16,6 @@
 - include: launch_cloudformation.yml
   when: not wimpy_needs_elb
 
-- include: scaling_policies.yml
\ No newline at end of file
+- include: scaling_policies.yml
+
+- include: notify.yml
```
Comment (Hyunk3l) [inline tasks/deploy.yml]: No way to avoid it?

**Review by Hyunk3l — APPROVED** [review]
👏 😄 👍

File: defaults/main.yml (hunk 1 above)
Comment (fiunchinho) [reply defaults/main.yml]: Yes, I've tested and it seemed to work

File: tasks/deploy.yml (hunk 3 above)
Comment (fiunchinho) [reply tasks/deploy.yml]: What do you mean? The task inside `notify.yml` won't be executed if the github project is not passed as parameter

File: defaults/main.yml (hunk 2 above)
Comment (fiunchinho) [reply defaults/main.yml]: You are right. FIxed!

File: tasks/deploy.yml (hunk 3 above)
Comment (Hyunk3l) [reply tasks/deploy.yml]: I didn't see `when: wimpy_github_project_url != ''` xD sorry!

**Review by ismaelfernandezscmspain — COMMENTED** [review]
nice PR and be druid

File: defaults/main.yml [hunk 4]
```diff
@@ -118,6 +119,16 @@ wimpy_aws_elb_stickiness:
 
 wimpy_aws_instance_type: t2.small # EC2 instance type
 
+# Github
```
Comment (aramirez-es) [inline defaults/main.yml]: Is this an Ansible's module?
If not, don't you should use `wimpy_` prefix?

**Comment by aramirez-es** [comment]
It seems very interesting but I'm worry about what happens if the notification to GH fails. I mean, Ansible will output error if either of notification tasks fail. 

We can ignore those errors so that the output will success even notification cannot be performed.

What do you think guys?

**Comment by ismaelfernandezscmspain** [comment]
mmmm... when can GH notification fail?
- GH down
- URL malformed
- Wrong credentials
...
I think that the job must fail if notification fails, then you will have to review why

For the other hand, the notification mustn't be a problem for deploy new version

**Comment by fiunchinho** [comment]
Maybe it's probably better to ignore the notification error, otherwise, we could get an error even though the application was correctly deployed.

**Review by aramirez-es — CHANGES_REQUESTED** [review]
Please, prefix `github` vars with `wimpy_` and ignore fails publishing statuses to GH.

File: tasks/create_github_deployment.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      Authorization: "Token {{ wimpy_github_token }}"
+      Content-Type: "application/json"
+    body:
+      state: "pending"
+      target_url: ""
+      description: "Deployment started"
+    body_format: json
+    status_code: 201
+    ignore_errors: yes
+    when: wimpy_github_deployment_out|succeeded
```
Comment (fiunchinho) [inline tasks/create_github_deployment.yml:33]: Ignoring errors and executing only if the task creating the deployment succeeded

**Comment by fiunchinho** [comment]
Fixed! Please, @aramirez-es take a look

**Comment by fiunchinho** [comment]
Also, I rebased from master

**Review by aramirez-es — APPROVED** [review]

**Pull request merged**